tests/quickinspectorpickingtest.cpp

```diff
@@ -0,0 +1,208 @@
+*/
+
+#include <config-gammaray.h>
+
+#include <plugins/quickinspector/quickinspectorinterface.h>
+#include <probe/hooks.h>
+#include <probe/probecreator.h>
+#include <core/probe.h>
+#include <core/toolmanager.h>
+#include <common/paths.h>
+#include <common/objectbroker.h>
+#include <common/remoteviewinterface.h>
+#include <common/remoteviewframe.h>
+
+#include <3rdparty/qt/modeltest.h>
+
+#include <QtTest/qtest.h>
+
+#include <QQuickView>
+#include <QItemSelectionModel>
+#include <QRegExp>
+#include <QSignalSpy>
+
+#if QT_VERSION < QT_VERSION_CHECK(5, 5, 0)
+Q_DECLARE_METATYPE(QItemSelection)
+#endif
+
+using namespace GammaRay;
+
+class QuickInspectorPickingTest : public QObject
+{
+    Q_OBJECT
+private:
+    void createProbe()
+    {
+        Paths::setRelativeRootPath(GAMMARAY_INVERSE_BIN_DIR);
+        qputenv("GAMMARAY_ProbePath", Paths::probePath(GAMMARAY_PROBE_ABI).toUtf8());
+        qputenv("GAMMARAY_ServerAddress", GAMMARAY_DEFAULT_LOCAL_TCP_URL);
+        Hooks::installHooks();
+        Probe::startupHookReceived();
+        new ProbeCreator(ProbeCreator::Create);
+        QTest::qWait(1); // event loop re-entry
+    }
+
+    void triggerSceneChange()
+    {
+        QTest::keyClick(view, Qt::Key_Right);
+        QTest::qWait(20);
+        QTest::keyClick(view, Qt::Key_Left);
+        QTest::qWait(20);
+        QTest::keyClick(view, Qt::Key_Right);
+    }
+
+    bool waitForSignal(QSignalSpy *spy, bool keepResult = false)
+    {
+        if (spy->isEmpty())
+            spy->wait(1000);
+        bool result = !spy->isEmpty();
+        if (!keepResult)
+            spy->clear();
+        return result;
+    }
+
+    bool showSource(const QString &sourceFile)
+    {
+        QSignalSpy renderSpy(view, SIGNAL(frameSwapped()));
+        Q_ASSERT(renderSpy.isValid());
+
+        view->setSource(QUrl(sourceFile));
+        view->show();
+        exposed = QTest::qWaitForWindowExposed(view);
+        if (!exposed)
+            qWarning()
+                <<
+            "Unable to expose window, probably running tests on a headless system - ignoring all following render failures.";
+
+
+        // wait at least two frames so we have the final window size with all render loop/driver combinations...
+        QTest::qWait(20);
+        waitForSignal(&renderSpy);
+        view->update();
+        return !exposed || waitForSignal(&renderSpy);
+    }
+
+private slots:
+    void initTestCase()
+    {
+        qRegisterMetaType<QItemSelection>();
+    }
+
+    void init()
+    {
+        createProbe();
+
+        // we need one view for the plugin to activate, otherwise the model will not be available
+        view = new QQuickView;
+        view->setResizeMode(QQuickView::SizeViewToRootObject);
+        QTest::qWait(1); // event loop re-entry
+
+        itemModel = ObjectBroker::model(QStringLiteral("com.kdab.GammaRay.QuickItemModel"));
+        QVERIFY(itemModel);
+        new ModelTest(itemModel, view);
```

Comment:
The model testing code here and below looks like a copy/paste leftover we probably don't need here?